ros/processors/report_processor.py

```diff
@@ -0,0 +1,112 @@
+import json
+import requests
+import logging
+import tarfile
+from io import BytesIO
+from http import HTTPStatus
+from confluent_kafka import Consumer
+from ros.lib.host_inventory import fetch_host_from_inventory
+from ros.app import app, db
+from ros.config import INSIGHTS_KAFKA_ADDRESS
+from ros.models import PerformanceProfile
+
+LOG = logging.getLogger(__name__)
+running = True
+
+
+consumer = Consumer({
+    'bootstrap.servers': INSIGHTS_KAFKA_ADDRESS,
+    'group.id': 'ros-consumers',
+    "enable.auto.commit": False
+})
+
+
+class ReportProcessor:
+    def init_consumer(self):
+        consumer.subscribe(['platform.upload.resource-optimization'])
+        while running:
+            msg = consumer.poll(1.0)
+            if msg is None:
+                continue
+            if msg.error():
+                print(msg.error())
+                continue
+            try:
+                self.msg = json.loads(msg.value().decode("utf-8"))
+                self.handle_message()
+            except json.decoder.JSONDecodeError:
+                print(f"Unable to decode kafka message - {msg.value()}")
+            except Exception as err:
+                print(f"An error occurred during message processing - {err}")
+            finally:
+                consumer.commit()
+
+        consumer.close()
+
+    def handle_message(self):
+        self.report_url = self.msg.get('url', None)
+        metadata = self.msg.get('metadata', None)
+        if not metadata:
+            return None
+        insights_id = metadata['insights_id']
+        if not insights_id:
+            return None
```

Comment:
```suggestion
        if not metadata or not metadata.get('insights_id'):
            return None
```
There is no any log statement or difference in return values so will it be good to handle both conditions in one if statement?

Reply:
:+1: